test/concentrate/vehicle_position_test.exs

```diff
@@ -23,5 +58,268 @@ defmodule Concentrate.VehiclePositionTest do
+          sources: MapSet.new(["busloc"])
+        )
+
+      swiftly_later =
+        new(
+          last_updated: 3,
+          latitude: 3,
+          longitude: 3,
+          trip_id: "swiftly_trip",
+          route_id: "swiftly_route",
+          sources: MapSet.new(["swiftly"])
+        )
+
+      expected =
+        new(
+          last_updated: 2,
+          latitude: 2,
+          longitude: 2,
+          trip_id: "swiftly_trip",
+          route_id: "swiftly_route",
+          sources: MapSet.new(["busloc", "swiftly"]),
+          data_discrepancies: [
+            %DataDiscrepancy{
+              attribute: :trip_id,
+              sources: [
+                %{id: "swiftly", value: "swiftly_trip"},
+                %{id: "busloc", value: "busloc_trip"}
+              ]
+            },
+            %DataDiscrepancy{
+              attribute: :route_id,
+              sources: [
+                %{id: "swiftly", value: "swiftly_route"},
+                %{id: "busloc", value: "busloc_route"}
+              ]
+            }
+          ]
+        )
+
+      expected_later =
+        new(
+          last_updated: 3,
+          latitude: 3,
+          longitude: 3,
+          trip_id: "swiftly_trip",
+          route_id: "swiftly_route",
+          sources: MapSet.new(["busloc", "swiftly"]),
+          data_discrepancies: [
+            %DataDiscrepancy{
+              attribute: :trip_id,
+              sources: [
+                %{id: "busloc", value: "busloc_trip"},
+                %{id: "swiftly", value: "swiftly_trip"}
+              ]
+            },
+            %DataDiscrepancy{
+              attribute: :route_id,
+              sources: [
+                %{id: "busloc", value: "busloc_route"},
+                %{id: "swiftly", value: "swiftly_route"}
+              ]
+            }
+          ]
+        )
+
+      assert Mergeable.merge(swiftly, non_swiftly) == expected
+      assert Mergeable.merge(non_swiftly, swiftly) == expected
+      assert Mergeable.merge(swiftly_later, non_swiftly) == expected_later
+      assert Mergeable.merge(non_swiftly, swiftly_later) == expected_later
+    end
+
+    test "merge/2 takes the other trip and route values if the swiftly values are nil" do
+      swiftly =
+        new(
+          last_updated: 1,
+          latitude: 1,
+          longitude: 1,
+          trip_id: nil,
+          route_id: nil,
+          sources: MapSet.new(["swiftly"])
+        )
+
+      non_swiftly =
+        new(
+          last_updated: 2,
+          latitude: 2,
+          longitude: 2,
+          trip_id: "busloc_trip",
+          route_id: "busloc_route",
+          sources: MapSet.new(["busloc"])
+        )
+
+      expected =
+        new(
+          last_updated: 2,
+          latitude: 2,
+          longitude: 2,
+          trip_id: "busloc_trip",
+          route_id: "busloc_route",
+          sources: MapSet.new(["busloc", "swiftly"]),
+          data_discrepancies: [
```

Comment:
One of the values being `nil` and the other having a value probably shouldn't be a discrepancy?

Reply:
Discussed this condition in person. We'll want to keep watch for false positives around trip boundaries.